cloudferrylib/os/identity/keystone.py

```diff
@@ -213,6 +214,8 @@ def deploy(self, info):
         self._upload_user_tenant_roles(roles, users, tenants)
         LOG.info("Done")
 
+    @retrying.retry(expected_exceptions=[ks_exceptions.Unauthorized],
+                    reraise_original_exception=True)
```

Comment:
Aa In run method of retrying.py, I have configured max_attempt as value of retry as in conf file. So it will not run infinite time.
If retry is in conf file - retry that many time
Else - run only 1 time.
Also, I have handled it in run method because it can not be handled in the decorator as cfglib is not initialized at class level and for the we need to use Global variable which is against the standard.
Please have a look on retrying.py too. Retry will not continue to infinite.